### PR TITLE
Script to validate directories changed in operator submission

### PR DESCRIPTION
**Pull request by MarcinGinszt**
Related PR: https://github.com/redhat-openshift-ecosystem/operator-pipelines/pull/12

### CODE REVIEW

File: operatorcert/utils.py [hunk 1]
```diff
@@ -21,3 +21,18 @@ def find_file(
         if new_path.exists() and new_path.is_file():
             return new_path
     return None
+
+
+def str_color(color, data):
```
Comment (Allda) [inline operatorcert/utils.py:26]: There is already a colored logger so I guess we don't have to re-implement a wheel again: https://pypi.org/project/coloredlogs/

Although I am not really sure if this is something that we need to have in our pipelines. What's an additional value?

File: operatorcert/__init__.py [hunk 2]
```diff
@@ -145,3 +146,55 @@ def ocp_version_info(bundle_path: pathlib.Path, pyxis_url: str) -> Dict:
...
+    organization: str, repository: str, base_branch: str, pr_head_label: str
+) -> List[str]:
+    """
+    Get the list of files modified in the PR against the base branch.
+    """
+    compare_changes_url = (
+        f"https://api.github.com/repos/{organization}/{repository}"
+        f"/compare/{base_branch}...{pr_head_label}"
+    )
+    pr_data = requests.get(compare_changes_url).json()
```
Comment (Allda) [inline operatorcert/__init__.py:181]: Use raise_for_status()

File: operatorcert/__init__.py [hunk 3]
```diff
@@ -145,3 +146,55 @@ def ocp_version_info(bundle_path: pathlib.Path, pyxis_url: str) -> Dict:
...
+    )
+
+    wrong_changes = False
+    for file_path in changed_files:
+        if file_path.startswith(path):
+            logging.info(str_color("green", f"Change path ok: {file_path}"))
+            continue
+        else:
+            logging.error(str_color("red", f"Wrong change path: {file_path}"))
+            wrong_changes = True
```
Comment (Allda) [inline operatorcert/__init__.py:217]: break?

Comment (MarcinGinszt) [reply operatorcert/__init__.py:217]: I'd like to log all of the wrong paths

File: operatorcert/utils.py (hunk 1 above)
Comment (MarcinGinszt) [reply operatorcert/utils.py:26]: The only value is color :slightly_smiling_face: 
I can remove it, I only wanted to make the output that is important for users to see more visible.

File: operatorcert/entrypoints/verify_changed_dirs.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+def main() -> None:
+    logging.basicConfig(stream=sys.stdout, level="INFO", format="%(message)s")
+
+    parser = argparse.ArgumentParser(
+        description="Determines the OCP version under test."
+    )
+    parser.add_argument("--operator_name", help="Unique name of the operator package")
+    parser.add_argument(
+        "--operator_version", help="Operator version in the semver format"
```
Comment (amisstea) [inline operatorcert/entrypoints/verify_changed_dirs.py:20]: Shouldn't this be the bundle version instead?

File: operatorcert/utils.py (hunk 1 above)
Comment (MarcinGinszt) [reply operatorcert/utils.py]: I removed the colorful logs. Maybe eventually, one day we'll add them for all the scripts.

File: operatorcert/__init__.py (hunk 2 above)
Comment (MarcinGinszt) [reply operatorcert/__init__.py]: done

File: operatorcert/entrypoints/verify_changed_dirs.py (hunk 4 above)
Comment (MarcinGinszt) [reply operatorcert/entrypoints/verify_changed_dirs.py]: done

File: operatorcert/__init__.py (hunk 3 above)
Comment (Allda) [reply operatorcert/__init__.py:214]: ok, makes sense

**Review by Allda — APPROVED** [review]

**Pull request merged**